### PR TITLE
Updates test file to use table-driven tests

### LINKED ISSUE #2: Start creating unity tests
By now we only have a single test doing a request. We need to create a structure to make easier for the mainteners to create new cases. Probably using table driven tests would be a good option to keep it simple, but robust.

a few links as reference:
https://yourbasic.org/golang/table-driven-unit-test/
https://www.codementor.io/@cyantarek15/how-table-driven-tests-makes-writing-unit-tests-exciting-and-fun-in-go-15g1wzdf7g
https://dev.to/boncheff/table-driven-unit-tests-in-go-407b

### PR DESCRIPTION
**Pull request by ArmoryNode**
Test definitions now follow the same pattern using the `Test` struct.
Fixes #2

### CODE REVIEW

File: fluentrequest_test.go [hunk 1]
```diff
@@ -3,105 +3,154 @@ package fluentrequest
...
-		Run()
+	var deserializedBody Body
 
-	body, _ := ioutil.ReadAll(resp.Body)
+	json.Unmarshal(responseBody, &deserializedBody)
 
 	assert.NoError(t, err)
-	assert.Equal(t, expectedPATCHBody, string(body))
-	assert.Equal(t, http.StatusOK, resp.StatusCode)
+	assert.Equal(t, want.responseBody, deserializedBody)
```
Comment (vitorestevam) [inline fluentrequest_test.go]: I was doing some experiments trying to turn our tests more maintenable and just found out the 
assert.JSONEq what is able to assert jsons by strings. 

Going into the code i just saw it does the same Unmarshal as you are doing. 

Using it we can build our Test object with strings in a very simplier and more legible way and removing the unmarshal of result validation

``` go 
{
	name:   "Test POST request",
	method: http.MethodPost,
	url:    "https://jsonplaceholder.typicode.com/todos/",
	body: ` {"Id": 201, "UserId": 2, "Title": "foo", "Body": "bar", "Completed": true`},
	want: ResponseResult{
		statusCode: http.StatusCreated,
		responseBody:  ` {"Id": 201, "UserId": 2, "Title": "foo", "Body": "bar", "Completed": true`}
	},
},
```

Comment (vitorestevam) [reply fluentrequest_test.go]: BTW you did a great job, thanks! This is just a suggestion. What do you think about it?

Comment (ArmoryNode) [reply fluentrequest_test.go]: That's actually perfect, thanks for the suggestion! I'm still new to Go, so I appreciate the suggestions.🙂I'll get the updates in.

File: fluentrequest_test.go [hunk 2]
```diff
@@ -2,106 +2,102 @@ package fluentrequest
...
-		Run()
-
-	body, _ := ioutil.ReadAll(resp.Body)
-
-	assert.NoError(t, err)
-	assert.Equal(t, expectedPUTBody, string(body))
-	assert.Equal(t, http.StatusOK, resp.StatusCode)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			createRequest(t, tt.method, tt.url, tt.body, tt.want)
```
Comment (vitorestevam) [inline fluentrequest_test.go]: Looks kinda strange having a function like this in a package of Chainned function calls. I Would move the test content (lines 85 to 102) to inside the t.Run directly. Also, maybe we will create tests for request witthout all propeties passed, so the function createRequest will not be able to be used.

After this, EXCELLENT work here! i'm very anxious to merge this PR!

Comment (ArmoryNode) [reply fluentrequest_test.go]: That's a really good point. Also, I just noticed that I didn't actually add a way to customize the header in the test. I'll add both of the fixes now.

File: fluentrequest_test.go [hunk 3]
```diff
@@ -2,106 +2,102 @@ package fluentrequest
...
-		"id":     1,
-		"title":  "delectus aut autem",
-		"body":   "bar",
-		"userId": 1,
-	})
-
-	header := http.Header{
-		"Content-Type": {"application/json; charset=UTF-8"},
+func TestRequest(t *testing.T) {
+	tests := []Test{
```
Comment (vitorestevam) [inline fluentrequest_test.go]: As this Test struct is being used only in this test preffer defining it here, not in the open scope. The same for ResponseResult

```suggestion
	tests := []struct {
		name   string
		method string
		url    string
		body   string
		want   struct {
			responseBody string
			statusCode   int
		}
	}{...}
```

Comment (ArmoryNode) [reply fluentrequest_test.go]: I'll make sure to include this in my changes too 👍🏼

**Review by vitorestevam — APPROVED** [review]
looks perfect, thanks! Feel free to merge it :D

**Comment by ArmoryNode** [comment]
Awesome! Unfortunately, I don't have permission to merge pull requests. 🙂

**Comment by vitorestevam** [comment]
> Awesome! Unfortunately, I don't have permission to merge pull requests. 🙂

Oh, ok! i'm still learning about PRs and stuff 😅 merged!

**Pull request merged**